pkg/runtime/task_resource_provider.go

```diff
@@ -3,26 +3,59 @@ package runtime
 import (
 	"github.com/flyteorg/flyteadmin/pkg/runtime/interfaces"
 	"github.com/flyteorg/flytestdlib/config"
+	"k8s.io/apimachinery/pkg/api/resource"
 )
 
 const taskResourceKey = "task_resources"
 
 var taskResourceConfig = config.MustRegisterSection(taskResourceKey, &TaskResourceSpec{})
 
 type TaskResourceSpec struct {
-	Defaults interfaces.TaskResourceSet `json:"defaults"`
-	Limits   interfaces.TaskResourceSet `json:"limits"`
+	Defaults         interfaces.DeprecatedTaskResourceSet `json:"defaults"`
+	DefaultResources interfaces.TaskResourceSet           `json:"defaultResources"`
+	Limits           interfaces.DeprecatedTaskResourceSet `json:"limits"`
+	LimitResources   interfaces.TaskResourceSet           `json:"limitResources"`
```

Comment:
Does that mean the administrator will need to migrate configs?

Reply:
talked offline, we don't need the migration path... we can just change the types from string to resource.Quantity....